### PR TITLE
Notes feature

### PR DESCRIPTION
**Pull request by JanaHoch**
New feature: add short description for each record

### CODE REVIEW

File: app/src/main/res/values-ru/strings.xml [hunk 1]
```diff
@@ -131,5 +131,6 @@
   <string name="operations">Операции</string>
 
   <string name="start_earlier_end">Дата конца не может быть раньше даты начала</string>
+    <string name="notes">примечания</string>
```
Comment (yev-kanivets) [inline app/src/main/res/values-ru/strings.xml:134]: Nit: The spacing is slightly off, and the first letter should be capitalized.

File: app/src/main/res/values-pt-rBR/strings.xml [hunk 2]
```diff
@@ -146,4 +146,5 @@
 
     <string name="title_summary_record" translatable="false">%1$s (%2$d)</string>
     <string name="start_earlier_end">A data final não pode ser anterior a data inicial</string>
+    <string name="notes">anotações</string>
```
Comment (yev-kanivets) [inline app/src/main/res/values-pt-rBR/strings.xml:149]: Nit: The capitalization is also needed here.

File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/controller/external/Head.java [hunk 3]
```diff
@@ -10,8 +10,9 @@ public interface Head {
     String TIME = "time";
     String TITLE = "title";
     String CATEGORY = "category";
+    String NOTES = "notes";
     String PRICE = "price";
     String CURRENCY = "currency";
     String DELIMITER = ";";
-    int COLUMN_COUNT = 5;
+    int COLUMN_COUNT = 7;
```
Comment (yev-kanivets) [inline app/src/main/java/com/blogspot/e_kanivets/moneytracker/controller/external/Head.java:17]: 6?

File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/entity/RecordItem.kt [hunk 4]
```diff
@@ -4,7 +4,7 @@ sealed class RecordItem {
 
     data class Header(val date: String) : RecordItem()
 
-    data class Record(val title: String, val categoryName: String, val fullPrice: Double, val currency: String, val isIncome: Boolean) : RecordItem() {
-        constructor(record: com.blogspot.e_kanivets.moneytracker.entity.data.Record) : this(record.title, record.category?.name?.toString().orEmpty(), record.fullPrice, record.currency, record.isIncome)
+    data class Record(val title: String, val categoryName: String, val notes: String, val fullPrice: Double, val currency: String, val isIncome: Boolean) : RecordItem() {
+        constructor(record: com.blogspot.e_kanivets.moneytracker.entity.data.Record) : this(record.title, record.category?.name?.toString().orEmpty(), record.notes, record.fullPrice, record.currency, record.isIncome)
```
Comment (yev-kanivets) [inline app/src/main/java/com/blogspot/e_kanivets/moneytracker/entity/RecordItem.kt:8]: If you have existing records (before the database migration), they don't have notes (`null`), and the app crashes on the launch here. To fix the behavior you need to provide a default value (empty string?). To test the fix, you can add a few records on the current Google Play version, then install the one from this branch.

File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/account/edit/fragment/AccountOperationsFragment.kt [hunk 5]
```diff
@@ -59,10 +59,11 @@ class AccountOperationsFragment : BaseFragment() {
             val type = if (it.fromAccountId == account.id) Record.TYPE_EXPENSE else Record.TYPE_INCOME
             val title = constructRecordTitle(type, it)
             val category = Category(getString(R.string.transfer).toLowerCase())
+            val notes = "Account Transfer"
```
Comment (yev-kanivets) [inline app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/account/edit/fragment/AccountOperationsFragment.kt:62]: This should be localized and taken from `strings.xml`.

**Review by yev-kanivets — CHANGES_REQUESTED** [review]
Hey @JanaHoch, thanks for the great addition 🙏 

I've left a few comments that should be easy to fix, but don't hesitate to ping me if you need help. I may not be very responsive during work days, but I should have a few spare hours every weekend in the worst case.

Do you intend to add any other features? If yes, don't hesitate to create tickets in the original repo to provide some visibility. Also, we can plan a Google Play deployment to publish all features you added to the users.

File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/controller/external/Head.java (hunk 3 above)
Comment (JanaHoch) [reply app/src/main/java/com/blogspot/e_kanivets/moneytracker/controller/external/Head.java:17]: Hey I've fixed most of the other things. But here we will also be adding ACCOUNT_ID to export right. so after that COLUMN_COUNT should become 7 right. So should I make it six and take care of it during the merge or something else.

File: app/src/main/res/values-ru/strings.xml (hunk 1 above)
Comment (JanaHoch) [reply app/src/main/res/values-ru/strings.xml]: Fixed with [abfe703](https://github.com/xorum-io/open_money_tracker/pull/34/commits/abfe7038194b61e69fa5eb42c158a3bee0ca592a)

File: app/src/main/res/values-pt-rBR/strings.xml (hunk 2 above)
Comment (JanaHoch) [reply app/src/main/res/values-pt-rBR/strings.xml]: Fixed with [0669331](https://github.com/xorum-io/open_money_tracker/pull/34/commits/0669331f398ff184dd02ee2ef21904b2ddad6d2f)

File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/entity/RecordItem.kt (hunk 4 above)
Comment (JanaHoch) [reply app/src/main/java/com/blogspot/e_kanivets/moneytracker/entity/RecordItem.kt:8]: Fixed with [610cc48](https://github.com/xorum-io/open_money_tracker/pull/34/commits/610cc4834b30bc81351585fc0b4588bca33a69a2)

**Comment by JanaHoch** [comment]
I'm currently not working on any new features. But if i get some time there are few more things I'd like to improve.
Yeah it would be great to get a release after this PR is merged. can you also release on F-Droid. I have a de-googled phone.

**Comment by yev-kanivets** [comment]
@JanaHoch thanks for the changes 🙏 I'm going to merge `dev` into `master` now, which should trigger the publication of builds to F-Droid and Google Play.

**Pull request merged**